src/modules/service-worker-remover/index.ts

```diff
@@ -25,13 +25,13 @@ import { AMP_PREFETCHED_LINKS } from '../link-prefetch/constants';
 export class ServiceWorkerRemover {
   async installNoOpServiceWorker(): Promise<void> {
     // Taking over the document
-    self.addEventListener('install', function(e: ExtendableEvent) {
-      const { skipWaiting } = self as ServiceWorkerGlobalScope;
+    self.addEventListener('install', function (e: ExtendableEvent) {
+      const { skipWaiting } = (self as unknown) as ServiceWorkerGlobalScope;
```

Comment:
@kristoferbaxter could you explain to me the need for cast here? curious why there is no guarantee that self is a instance of `window`

Reply:
This was an updated version of the casting done by the previous code to fit TypeScript 4 (updated in this PR).

I plan to refactor this to avoid the casting.